impl/src/main/java/org/jboss/weld/injection/InterceptionFactoryImpl.java

```diff
@@ -108,6 +108,7 @@ public T createInterceptedInstance(T instance) {
                 .getInterceptionFactoryData(configurator != null ? configurator.complete() : annotatedType);
 
         if (!cached.isPresent()) {
+            used = true;
```

Comment:
I think we should move `used = true;` right after `throw InterceptorLogger.LOG.interceptionFactoryNotReusable()` and also remove L125.